glean_parser/parser.py

```diff
@@ -86,7 +84,14 @@ def _load_file(
     if not isinstance(schema_key, str):
         raise TypeError(f"Invalid schema key {schema_key}")
 
-    filetype = FILE_TYPES.get(schema_key)
+    filetype: Optional[str] = None
+    try:
+        filetype = schema_key.split("/")[-2]
+    except IndexError:
+        filetype = None
+
+    if filetype not in ("metrics", "pings"):
+        filetype = None
```

Comment:
Should the filetype be decided based on the schema, instead of hardcoding names? I know that this behaviour wasn't introduced here, so we can leave it be for now. Maybe we could file a bug for it?

Reply:
Yeah -- I think the `$schema` tag should define what's in the file, not hardcoded filenames -- that's how the JSON schema spec says things should work, and also, I like keeping the possibility open of having two `metrics.yaml` files in the same directory, even though none of our users seem to currently do that.

Reply:
Right! We weirdly do that for project EXTRACT: I think we use two metrics file there :o How is that even working?!

Reply:
Oh -- to be clear, the current code is based on the schema -- it pulls out the `metrics` part from `moz://mozilla.org/schemas/glean/metrics/2-0-0`.  I forgot EXTRACT was already relying on this.  Another good reason to keep this behavior.